File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    var len = arr.length;
    if (len % 2 != 0) {
        return arr.map((v, i) => {
            return (i % 2 == 0) ? v + n : v;
        });
    } else {
        return arr.map((v, i) => {
            return (i % 2 != 0) ? v + n : v;
        });
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.swift
import Foundation

func solution(_ arr:[Int], _ n:Int) -> [Int] {
    let cnt = arr.count
    var ans = [Int](repeating: 0, count: cnt)
    
    for i in 0..<cnt {
        ans[i] = arr[i] + (cnt % 2 + i % 2) % 2 * n
    }
    
    return ans
}